src/Compilers/VisualBasic/Portable/Parser/ParseInterpolatedString.vb

```diff
@@ -235,32 +235,23 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
                 closeBraceToken = DirectCast(CurrentToken, PunctuationSyntax)
                 GetNextToken(ScannerState.InterpolatedStringContent)
 
-            ElseIf CurrentToken.Kind = SyntaxKind.EndOfInterpolatedStringToken
+            ElseIf CurrentToken.Kind = SyntaxKind.EndOfInterpolatedStringToken Then
                 GetNextToken(ScannerState.VB)
 
                 closeBraceToken = DirectCast(HandleUnexpectedToken(SyntaxKind.CloseBraceToken), PunctuationSyntax)
             Else
                 ' Content rules will either resync at a } or at the closing ".
-                ResetCurrentToken(ScannerState.InterpolatedStringFormatString)
-
-                Dim skippedToken As SyntaxToken = Nothing
-
-                If CurrentToken.Kind = SyntaxKind.InterpolatedStringText Then
```

Comment:
This condition was never true, ```SyntaxKind.InterpolatedStringText``` is syntax node kind, a token cannot have this kind.